### ISSUE TITLE
make a tiny demo of keyup handler trouble

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#802]
Create the smallest standalone demo of the [keyup handler issue](https://github.com/ctm/mb2-doc/issues/801#issuecomment-1008289988).

Doing so may reveal a misunderstanding on my part. However, if it doesn't, I can ask about it on discord to figure out how to proceed.

This may take me a little while just because I haven't created a new Yew project in a long time and so I'll have to fiddle with trunk. OTOH, the documentation should walk me through it.

**Comment by ctm** [comment]
Ugh. Trunk doesn't support https.

**Comment by ctm** [comment]
Looks like I can use [simple-http-server](https://github.com/TheWaWaR/simple-http-server) and [Let's Encrypt to generate a certificate for localhost](https://letsencrypt.org/docs/certificates-for-localhost/). I don't know if that will be sufficient to trigger autofill, but it shouldn't be too hard to try.

**Comment by ctm** [comment]
Trying to use via:
    
    simple-http-server --index --cert localhost.pfx --certpass password -- dist
    
    No success.  I get the error message:
        localhost unexpectedly closed the connection.
    
    FWIW, I had to generate the pfx file with:
    
        openssl pkcs12 -export -out localhost.pfx -inkey localhost.key -in localhost.crt
    
    Initially I tried it with no password, but that gave me:
    
    thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Tls(Error { code: -25264, message: "MAC verification failed during PKCS12 import (wrong password?)" })', /Users/ctm/.cargo/registry/src/github.com-1ecc6299db9ec823/simple-http-server-0.6.1/src/main.rs:333:70
    
    I don't see any sort of debug option to help me figure out why the connection is closing, so I'll probably move on to some other server, but not this morning.

**Comment by ctm** [comment]
So, I got something working and managed to track this down to a [Chromium bug that's been around a long time](https://bugs.chromium.org/p/chromium/issues/detail?id=780834). It may also be [this bug](https://bugs.chromium.org/p/chromium/issues/detail?id=1214969). A few observations:

It doesn't happen on Firefox. That's actually something I should have checked immediately, but when this bug came in I was asleep and not thinking. If I had noticed that it didn't happen on Firefox, I probably would have checked the Chromium bug-tracker **much** sooner.

Using `gloo-console` I get the `Event` "inner event type" (my on phrase) on Brave, Chrome and Edge, but `CustomEvent` on Safari:

Brave, Chrome and Edge:
```
[src/lib.rs:9] ::std::format!("{:?}", v) = KeyboardEvent { obj: UiEvent { obj: Event { obj: Object { obj: JsValue(Event) } } } }
```

Safari:
```
[src/lib.rs:9] ::std::format!("{:?}", v) = – "KeyboardEvent { obj: UiEvent { obj: Event { obj: Object { obj: JsValue(CustomEvent) } } } }"
```

The `.type` property is `keyup` on all the Chromium browsers.

So, it looks to me like the auto-fill code creates (perhaps inadvertently) an event that isn't really what it claims to be and somehow that winds up getting picked up by the `onkeyup` handler. It seems hard to believe that `wasm-bindgen` (much less `yew`!) is doing anything wrong.

Unfortunately, I don't really have the chops to wrap this one up into something a non-Rust user can use to see the issue. In fact, I'm apt to forget some of the hoops I had to jump through to get everything packaged up and served (I eventually went with the [https crate](https://docs.rs/crate/https/latest).

Granted, part of the reason it took me so long to find this was my tiredness, but none of the web-searching (as opposed to searching the Chromium bug tracker) that I did turned up anything useful. As such, I'd like to help prevent someone else wasting time trying to track this down, so I'll probably squash my OnKeyUpDemo commits together, write a few notes (most of this comment will go in them) and make it a public repository.
